drivers/adc/ad7124/ad7124.c

```diff
@@ -76,29 +76,46 @@ int32_t ad7124_no_check_read_register(struct ad7124_dev *dev,
 	buffer[0] = AD7124_COMM_REG_WEN | AD7124_COMM_REG_RD |
 		    AD7124_COMM_REG_RA(p_reg->addr);
 
+	/*
+	 * If this is an AD7124_DATA register read, and the DATA_STATUS bit is set
+	 * in ADC_CONTROL, need to read 4, not 3 bytes for DATA with STATUS
+	 */
+	if ((p_reg->addr == AD7124_DATA_REG) &&
+	    (dev->regs[AD7124_ADC_Control].value & AD7124_ADC_CTRL_REG_DATA_STATUS)) {
+		add_status_length = 1;
```

Comment:
i understood the gist of this change;
but [the change] points out that this code should have been written in a neater fashion [when it was originally written];

nothing needs to change here; this change is fine as-is;
just a thought :)